plugin/documents.py

```diff
@@ -491,6 +491,11 @@ def on_text_command(self, command_name: str, args: Optional[dict]) -> Optional[T
         return None
 
     def on_post_text_command(self, command_name: str, args: Optional[Dict[str, Any]]) -> None:
+        if command_name == 'paste' and userprefs().format_on_paste:
+            # if we don't delay execution self.format_on_paste,
+            # on_text_changed_async will not pick up the change
+            # thus a call to self.purge_changes_async will not see new text changes.
```

Comment:
Might be enough to say:

```suggestion
            # ensure on_text_changed_async is triggered before self.purge_changes_async()
```